materialize-sql/type_mapping.go

```diff
@@ -24,15 +20,17 @@ type FlatType string
 
 // FlatType constants that are used by TypeMapper
 const (
-	ARRAY    FlatType = "array"
-	BINARY   FlatType = "binary"
-	BOOLEAN  FlatType = "boolean"
-	INTEGER  FlatType = "integer"
-	MULTIPLE FlatType = "multiple"
-	NEVER    FlatType = "never"
-	NUMBER   FlatType = "number"
-	OBJECT   FlatType = "object"
-	STRING   FlatType = "string"
+	ARRAY          FlatType = "array"
+	BINARY         FlatType = "binary"
+	BOOLEAN        FlatType = "boolean"
+	INTEGER        FlatType = "integer"
+	MULTIPLE       FlatType = "multiple"
+	NEVER          FlatType = "never"
+	NUMBER         FlatType = "number"
+	OBJECT         FlatType = "object"
+	STRING         FlatType = "string"
+	STRING_INTEGER FlatType = "string_integer"
```

Comment:
Adding these are separate `FlatType`'s helped clean up their handling. `STRING_INTEGER` and `STRING_NUMBER` are only generated for fields that aren't primary keys, so each materialization's type mapping configuration doesn't need to account for that.